File: Tugas12/Web_api/Pinjam.php
<?php
//Simpanlah dengan nama file : Pinjam.php
require_once 'database.php';
class Pinjam 
{
    private $db;
    private $table = 'pinjam';
    public $kodePinjam = "";
    public $kodeAnggota = "";
    public $kodeBuku = "";
    public $tglKembali = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kodePinjam(int $kodePinjam)
    {
        $query = "SELECT * FROM $this->table WHERE kodePinjam = $kodePinjam";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kodePinjam`,`kodeAnggota`,`kodeBuku`,`tglKembali`) VALUES ('$this->kodePinjam','$this->kodeAnggota','$this->kodeBuku','$this->tglKembali')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kodePinjam = '$this->kodePinjam', kodeAnggota = '$this->kodeAnggota', kodeBuku = '$this->kodeBuku', tglKembali = '$this->tglKembali' 
        WHERE idPinjam = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kodePinjam($kodePinjam): int
    {
        $query = "UPDATE $this->table SET kodePinjam = '$this->kodePinjam', kodeAnggota = '$this->kodeAnggota', kodeBuku = '$this->kodeBuku', tglKembali = '$this->tglKembali' 
        WHERE kodePinjam = $kodePinjam";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE idPinjam = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kodePinjam($kodePinjam): int
    {
        $query = "DELETE FROM $this->table WHERE kodePinjam = $kodePinjam";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Tugas12/Client_api/Pinjam.py
import requests
import json
class Pinjam:
    def __init__(self):
        self.__id=None
        self.__kodePinjam = None
        self.__kodeAnggota = None
        self.__kodeBuku = None
        self.__tglKembali = None
        self.__url = "http://localhost/MYLIBRARY/pinjam_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kodePinjam(self):
        return self.__kodePinjam
        
    @kodePinjam.setter
    def kodePinjam(self, value):
        self.__kodePinjam = value
    @property
    def kodeAnggota(self):
        return self.__kodeAnggota
        
    @kodeAnggota.setter
    def kodeAnggota(self, value):
        self.__kodeAnggota = value
    @property
    def kodeBuku(self):
        return self.__kodeBuku
        
    @kodeBuku.setter
    def kodeBuku(self, value):
        self.__kodeBuku = value
    @property
    def tglKembali(self):
        return self.__tglKembali
        
    @tglKembali.setter
    def tglKembali(self, value):
        self.__tglKembali = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kodePinjam(self, kodePinjam):
        url = self.__url+"?kodePinjam="+kodePinjam
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['idPinjam']
            self.__kodePinjam = item['kodePinjam']
            self.__kodeAnggota = item['kodeAnggota']
            self.__kodeBuku = item['kodeBuku']
            self.__tglKembali = item['tglKembali']
        return data
    def simpan(self):
        payload = {
            "kodePinjam":self.__kodePinjam,
            "kodeAnggota":self.__kodeAnggota,
            "kodeBuku":self.__kodeBuku,
            "tglKembali":self.__tglKembali
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kodePinjam(self, kodePinjam):
        url = self.__url+"?kodePinjam="+kodePinjam
        payload = {
            "kodePinjam":self.__kodePinjam,
            "kodeAnggota":self.__kodeAnggota,
            "kodeBuku":self.__kodeBuku,
            "tglKembali":self.__tglKembali
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kodePinjam(self,kodePinjam):
        url = self.__url+"?kodePinjam="+kodePinjam
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text